### PR TITLE
Don't report rude edits for using declaration changes

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/10043

This turned out really straight forward. Let me know there are more tests to cover.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs [hunk 1]
```diff
@@ -218,6 +211,171 @@ namespace N
                 "Delete [using System.Collections;]@22");
         }
 
+        [Fact]
+        public void Using_Delete_ChangesCodeMeaning()
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:215]: New tests need WorkItem for the linked issue. Not sure about updated tests.

Comment (davidwengier) [reply src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:215]: Really? I've only ever done that for bugs.

Comment (Youssef1313) [reply src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs:215]: @davidwengier I thought this is considered a bug :smile:
If it's more like a feature request implementation, probably no need.

**Comment by davidwengier** [comment]
@tmat in case you hadn't seen this

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs [hunk 2]
```diff
@@ -10500,6 +10500,89 @@ .maxstack  1
...
+            // Pretend there was an update to C.E to ensure we haven't invalidated the test
+
+            var diffError = compilation1.EmitDifference(
+              generation0,
+              ImmutableArray.Create(
+                  SemanticEdit.Create(SemanticEditKind.Update, e0, e1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+
+            diffError.EmitResult.Diagnostics.Verify(
+#if NETFRAMEWORK
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs:10556]: It'd be better if the test did not depend on the TFM the unit test assembly is built against. You should be able  to pass a specific TargetFramework to one of the CreateCompilation overloads.

**Review by tmat — APPROVED** [review]
Looks good modulo the ifdef in the test.

Let's also make the same change for VB - can be a separate PR.

Let's file an issue to follow up on in future (if necessary) to detect semantic changes caused by changed using directives and report warning on them.

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs [hunk 3]
```diff
@@ -267,8 +267,6 @@ static void CheckForExperimentStatus(ITextView textView, Document document)
                 roslynTrigger = new CompletionTrigger(CompletionTriggerKind.Snippets);
             }
 
-            var disallowAddingImports = _isDebuggerTextView ||
```
Comment (davidwengier) [inline src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs:270]: @genlu for completion changes.

This doesn't actually fix the issue, but at least the items show up in completion. Somehow the session properties still have DisallowAddingImports set to true, but I haven't been able to work out where that comes from yet. I set breakpoints everywhere that its set to true, and they don't get hit ¯\_(ツ)_/¯

**Comment by davidwengier** [comment]
Proof!

EnCAddImport

**Comment by davidwengier** [comment]
> Let's file an issue to follow up on in future (if necessary)

https://github.com/dotnet/roslyn/issues/52161

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs (hunk 3 above)
Comment (davidwengier) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs:270]: Nevermind, cleaned out my hive and this does fix it. Caches be caching I guess? ¯\\\_(ツ)_/¯

**Comment by davidwengier** [comment]
Gonna merge this because it's Friday for me, and we can sneak it into P2. Gen I'll follow up with any completion changes you want.

**Pull request merged**